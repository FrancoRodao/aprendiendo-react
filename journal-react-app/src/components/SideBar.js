import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { useHistory } from 'react-router'
import { startLogout } from '../redux/actions/authActions'
import { startAddNewNote } from '../redux/actions/notesActions'
import JournalEntries from './journal/JournalEntries'

export default function SideBar() {

    const dispatch = useDispatch()
    const history = useHistory()
    const userName = useSelector(state => state.auth.name)

    const handleAddEntry = () => {
        dispatch(startAddNewNote())
    }

    const handleLogout = () => {
        dispatch( startLogout() )
        history.push("/login")
    }

    return (
        <aside className="journal__sidebar">
            <div className="journal__sidebar-navbar">
                <h3 className="mt-5">
                    <span>{userName}</span>
                </h3>

                <button onClick={handleLogout} className="btn">Logout</button>
            </div>

            <div onClick={handleAddEntry} className="journal_new-entry">
                <i className="far fa-calendar-plus fa-5x"></i>
                <p className="mt-5">
                    New entry
                </p>
            </div>

            <JournalEntries />
        </aside>
    )
}
