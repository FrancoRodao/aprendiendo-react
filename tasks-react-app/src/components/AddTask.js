import React from 'react'

export default function AddTask({ handleSubmit, handleInputChange, values }) {
    
    return (
        <>
            <form onSubmit={handleSubmit}>
                <p>Title: </p> <input onChange={handleInputChange} name="title" />
                <p>Desc: </p> <input onChange={handleInputChange} name="desc" />
                <p>Done: </p> <input onChange={handleInputChange} name="done" type="checkbox" />
                <button type="submit">Add task</button>
            </form>

            <div>
            <p>Preview: </p>
            <p>Title: </p> <input readOnly value={values.title} />
            <p>Desc: </p> <input readOnly value={values.desc} />
            <p>Done: </p> <input readOnly type="checkbox" checked={values.done} />
            </div>
        </>
    )
}
