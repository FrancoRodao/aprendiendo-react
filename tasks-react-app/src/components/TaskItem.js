import React from 'react'

export const TaskItem = ({title, desc, done, dispatch, task})=>{
    
    const handleChange = ({target})=>{
        dispatch({
            type: "edit",
            payload: {
                ...task,
                //checkbox validation
                [target.name]: target.value === "on" ? target.checked : target.value
            }
        })
    }

    return (
        <div>
            <h3>{title}</h3>
            <p>{desc}</p>

            <p>done:</p> 
            <input type="checkbox" 
                        name="done" 
                        checked={done} 
                        onChange={handleChange}>
            </input>
        </div>
    )
}