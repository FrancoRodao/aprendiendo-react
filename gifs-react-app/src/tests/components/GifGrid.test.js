import React from 'react'
import useFetchGifs from '../../hooks/useFetchGifs'
import { shallow } from "enzyme"
import { GifGrid } from "../../components/GifGrid";
jest.mock('../../hooks/useFetchGifs');

describe('GifGrid test component', () => {

    beforeEach(()=>{
        jest.clearAllMocks()
    })

    let category = "coding"
    useFetchGifs.mockReturnValue({
        gifs: [],
        loading: true
    })
    let wrapper = shallow(<GifGrid category={category}/>)

    test('should display snapshot correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('should show items when gifs are loaded', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa'
        },
        {
            id: '123',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa'
        }]

        useFetchGifs.mockReturnValue({
            gifs,
            loading: false
        })
        const wrapper = shallow(<GifGrid category={category}/>)
        expect(wrapper.find("GifGridItem").length).toBe(gifs.length)
    })
    
})
