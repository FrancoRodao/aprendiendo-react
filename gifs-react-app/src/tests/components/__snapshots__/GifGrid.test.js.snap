// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid test component should display snapshot correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    coding
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
